lib/Transforms/Scalar/ScalarReplAggregatesHLSL.cpp

```diff
@@ -75,6 +76,28 @@ STATISTIC(NumReplaced, "Number of allocas broken up");
 
 namespace {
 
+class RPOTHelper {
+public:
+  using RPOTIndexMap = DenseMap<const BasicBlock *, unsigned>;
+
+  RPOTIndexMap &getIndexMap(Function *F) {
+    if (RPOTCache.find(F) == RPOTCache.end()) {
+      ReversePostOrderTraversal<Function *> RPOT(F);
+      RPOTIndexMap RPOTOrder;
+      unsigned Count = 0;
+      for (BasicBlock *BB : RPOT) {
+        RPOTOrder[BB] = Count++;
+      }
+      RPOTCache[F] = RPOTOrder;
+    }
+    return RPOTCache[F];
+  }
+
+private:
+  using RPOTFunctionMap = DenseMap<const Function *, RPOTIndexMap>;
```

Comment:
Now that this type is totally private and no uses of it are required elsewhere, naming this isn't really necessary.  I'm not asking for a change, just pointing out that naming it is really not necessary for the original reason I asked anymore - the class serves that purpose now.